src/workspaces/list/WorkspaceStateCell.tsx

```diff
@@ -0,0 +1,149 @@
+  const description = attributes?.description;
+  const errorMessage = workspace.errorMessage;
+  switch (state) {
+    case 'Deleting':
+      return <WorkspaceDeletingCell />;
+    case 'DeleteFailed':
+      return <WorkspaceFailedCell state={state} errorMessage={errorMessage} />;
+    case 'Deleted':
+      return <WorkspaceDeletedCell />;
+    case 'Cloning':
+      return <WorkspaceCloningCell />;
+    case 'CloningContainer':
+      return <WorkspaceCloningCell />;
+    case 'CloningFailed':
+      return <WorkspaceFailedCell state={state} errorMessage={errorMessage} />;
+    default:
+      return <WorkspaceDescriptionCell description={description} />;
+  }
+};
+
+const WorkspaceDeletingCell = (): ReactNode => (
+  <WorkspaceStatusPill iconShape='syncAlt' rotateIcon color={colors.danger}>
+    Workspace deletion in progress
+  </WorkspaceStatusPill>
+);
+
+const WorkspaceCloningCell = (): ReactNode => (
+  <WorkspaceStatusPill iconShape='syncAlt' rotateIcon color={colors.success}>
+    Workspace cloning in progress
+  </WorkspaceStatusPill>
+);
+
+const WorkspaceStatusPill = (props: {
+  iconShape: IconId;
+  rotateIcon?: boolean;
+  color: (number?) => any;
+  children: ReactNode;
+}) => {
+  return (
+    <div
+      style={{
+        backgroundColor: props.color(0.25),
+        width: 'fit-content',
+        paddingTop: '0.25rem',
+        paddingBottom: '0.25rem',
+        paddingLeft: '1rem',
+        paddingRight: '1rem',
+        fontWeight: 500,
+        textAlign: 'center',
+        borderRadius: '1rem',
+        fontSize: '0.875rem',
+      }}
+    >
+      {icon(props.iconShape, {
+        size: 16,
+        style: {
+          animation: props.rotateIcon ? 'rotation 2s infinite linear' : undefined,
+          color: props.color(),
+          textAlign: 'center',
+        },
+      })}
+      <span style={{ margin: '0.5rem' }}>{props.children}</span>
+    </div>
+  );
+};
+
+interface WorkspaceFailedCellProps {
+  state: 'DeleteFailed' | 'CloningFailed';
+  errorMessage?: string;
+}
+
+const WorkspaceFailedCell = (props: WorkspaceFailedCellProps): ReactNode => {
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+
+  const failureMsg = props.state === 'DeleteFailed' ? 'Error deleting workspace' : 'Workspace clone unsuccessful';
+
+  return (
+    <WorkspaceStatusPill iconShape='warning-standard' color={colors.danger}>
+      {failureMsg}
+      {props.errorMessage ? (
+        // eslint-disable-next-line jsx-a11y/anchor-is-valid
+        <Link
+          onClick={() => setShowDetails(true)}
+          style={{ fontSize: 14, marginRight: '0.5rem', marginLeft: '0.5rem' }}
+        >
+          See error details.
+        </Link>
+      ) : null}
+      {showDetails ? (
+        <Modal width={800} title={failureMsg} showCancel={false} showX onDismiss={() => setShowDetails(false)}>
+          <ErrorView error={props.errorMessage ?? 'No error message available'} />
+        </Modal>
+      ) : null}
```

Comment:
Nit: conditional rendering can be slightly more concise with `condition && element` vs `condition ? element : null`.

```suggestion
      {showDetails && (
        <Modal width={800} title={failureMsg} showCancel={false} showX onDismiss={() => setShowDetails(false)}>
          <ErrorView error={props.errorMessage ?? 'No error message available'} />
        </Modal>
      )}
```